plugin/init.lua

```diff
@@ -52,12 +52,15 @@ end
 ---@param file_name string
 ---@param type string
 ---@param opt_name string?
+---@param date_fmt string?
 ---@return string
-local function get_file_path(file_name, type, opt_name)
+local function get_file_path(file_name, type, opt_name, date_fmt)
 	if opt_name then
 		file_name = opt_name
 	end
-	return string.format("%s%s" .. separator .. "%s.json", pub.save_state_dir, type, file_name:gsub(separator, "+"))
+	date_fmt = date_fmt and " " .. wezterm.strftime(date_fmt) or ""
+	return string.format("%s%s" .. separator .. "%s%s.json", pub.save_state_dir, type, file_name:gsub(separator, "+"),
+		date_fmt)
```

Comment:
This can be syntactically cryptic for someone without much lua experience, so maybe changing this to something similar to the following would be easier to read:

```suggestion
	if date_fmt then
	    return string.format("%s%s" .. separator .. "%s %s.json", pub.save_state_dir, type, file_name:gsub(separator, "+"),
		wezterm.strftime(date_fmt))
	else
		    return string.format("%s%s" .. separator .. "%s.json", pub.save_state_dir, type, file_name:gsub(separator, "+"))
  end
```